doc/spec_dataValidityPropagation.dox

```diff
@@ -85,6 +87,49 @@ See @ref spec_execptionHandling.
 
 - 3.1 The decorators which manipulate the data fault counter are responsible for counting up and down in pairs, such that the counter goes back to 0 if all data is ok, and never becomes negative.
 
+4. Circular dependencies
+------------------------
+
+If modules have circular dependencies, the algorithm described in \ref dataValidity_1 "section 1" leads to a self-excited loop: Once the DataValidity::invalid flag has made a full circle, there is always at least on input with invalid data in each module and you can't get rid of it any more. To break this circle, the following additional behaviour is implemented:
+
+### 4.1 General behaviour
+
+- \anchor dataValidity_4_1_1 4.1.1 Inputs which are part of a circular dependency are marked as _circular input_.[\ref dataValidity_test_TestCircularInputDetection "T"]
+    - \anchor dataValidity_4_1_1_1 4.1.1.1 Inputs which are coming from other applicatiation modules
+ which are not part of the circle, from the control system module or from device modules are considered _external inputs_.[\ref dataValidity_test_TestCircularInputDetection "T" (only CS module)]
```

Comment:
Also forcing the data validity to faulty programmatically in an ApplicationModule is considered an external input, isn't it?

Reply:
Whoops, this is missing in the implementation. Nice catch!

Reply:
Updated the spec, put a test and implemented it in the MetaDataPropagatingRegisterAccessor.